numpy/lib/tests/test_function_base.py

```diff
@@ -347,8 +347,11 @@ def test_weights(self):
         desired = np.array([3., 6.])
         assert_almost_equal(actual, desired)
 
-        # This should raise an error. Can we test for that ?
-        # assert_equal(average(y1, weights=w1), 9./2.)
+        # weights and input have different shapes but no axis is specified
+        with assert_raises_regex(
```

Comment:
Nitpick, but for new code one might as well use to the more current `with pytest.raises(TypeError, match="..."):`

Reply:
@mhvk Done.